vllm/attention/backends/xformers.py

```diff
@@ -226,35 +243,41 @@ def forward(
                                                   self.num_queries_per_kv,
                                                   value.shape[-1])
 
-                output = self._run_memory_efficient_xformers_forward(
-                    query, key, value, attn_metadata)
+                out = self._run_memory_efficient_xformers_forward(
+                    query, key, value, prefill_meta).squeeze(0)
+                assert out.shape == output[:num_prefill_tokens].shape
+                output[:num_prefill_tokens] = out
             else:
                 # prefix-enabled attention
                 # TODO(Hai) this triton kernel has regression issue (broke) to
                 # deal with different data types between KV and FP8 KV cache,
                 # to be addressed separately.
-                output = PagedAttention.forward_prefix(
+                out = PagedAttention.forward_prefix(
                     query,
                     key,
                     value,
                     key_cache,
                     value_cache,
-                    attn_metadata.block_tables,
-                    attn_metadata.subquery_start_loc,
-                    attn_metadata.prompt_lens_tensor,
-                    attn_metadata.context_lens,
-                    attn_metadata.max_subquery_len,
+                    prefill_meta.block_tables,
```

Comment:
@rkooo567 you mentioned this kernel is really slow right? should we show some warning if someone uses this backend with chunked prefills?

Reply:
Hmm feel like that's a bit weird because there's no alternative option. I think we should probably just mark it as experimental feature for now?

Reply:
I can also do some additional e2e benchmark. When I ran it with 7B, the perf was actually not very different